confgen/api_internal.go

```diff
@@ -486,6 +486,7 @@ func computeVolumeSetChange(oldConfMap conf.ConfMap, newConfMap conf.ConfMap) (t
 	_, _, newLocalVolumeMap, _, _, err = fetchVolumeInfo(newConfMap)
 	if nil != err {
 		err = fmt.Errorf("In newConfMap: %v", err)
+		return
```

Comment:
Does this work if there are no volumes and no VolumeGroups in the configuration?

Reply:
Hmmm.  By my reading of the code, lack of volumes and volume groups will not cause `fetchVolumeInfo()` to return an error.  If it did, we would have run into the problem with the same call 6 lines earlier.

Again by my reading of the code, it looks like `computeVolumeSetChange()` works fine if the `newLocalVolumeMap` or `oldLocalVolumeMap` is empty.

We're not using `computeVolumeSetChange()` anyway so I'm not going to spend time testing it.  I just added a return that seemed to be missing, though looking at the code we would wind up returning from the bottom of the function with `err != nil` so this is not actually a functional change.

Reply:
In reference to "It seems like this function should be reorganized to take advantage of IsVolumeGroupSharedSMB() and IsVolumeGroupSharedNFS()":

I'm not sure that that would improve readability or maintainability.  I'd have to add a loop over volume groups and then construct an SMB volume group list.